File: leet_easy_c++/leet_283/leet_283a.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        stable_sort(nums.begin(), nums.end(), [](int a, int b){return a!=0 || b==0;});
        int i=0;
        while(i<nums.size() && nums[i]!=0) i++;
        reverse(nums.begin(), nums.begin()+i);
    }
};

File: leet_easy_java/leet_283/leet_283a.java
class Solution {
    public void moveZeroes(int[] nums) {
        int zero = nums.length - 1;
        for(int j=0, z = 0; j<zero; ){
            for(j=z; j<nums.length; j++){
                if(nums[j]==0){
                    z = j;
                    break;
                }
            }
            if(j >= zero) break;
            for(int i = z+1; i<=zero; i++){
                nums[i-1] = nums[i];    
            }   
            nums[zero] = 0;
            zero--;           
        }
    }
}